crates/bevy_pbr/src/meshlet/asset.rs

```diff
@@ -87,16 +110,44 @@ impl AssetSaver for MeshletMeshSaverLoad {
     type Asset = MeshletMesh;
     type Settings = ();
     type OutputLoader = Self;
-    type Error = bincode::Error;
+    type Error = MeshletMeshSaveOrLoadError;
 
     async fn save<'a>(
         &'a self,
         writer: &'a mut Writer,
         asset: SavedAsset<'a, Self::Asset>,
         _settings: &'a Self::Settings,
     ) -> Result<(), Self::Error> {
-        let bytes = bincode::serialize(asset.get())?;
+        writer
+            .write_all(&MESHLET_MESH_ASSET_VERSION.to_le_bytes())
+            .await?;
+
+        let mut bytes = Vec::new();
+        let mut sync_writer = FrameEncoder::new(&mut bytes);
+        bincode::serialize_into(&mut sync_writer, asset.get())?;
+        sync_writer.finish()?;
         writer.write_all(&bytes).await?;
+
         Ok(())
     }
 }
+
+#[derive(thiserror::Error, Debug)]
+pub enum MeshletMeshSaveOrLoadError {
```

Comment:
Maybe this could just be `MeshletSaveOrLoadError` to make it slightly less wordy? Are there any other assets that meshlets might want to load and save?

(I was originally going to suggest `MeshletIoError`, but `MeshletIoError::Io()` would be confusing, so yeah)